### PR TITLE
[BugFix] use database should check privilege of pipe

### LINKED ISSUE #34294: [pipe load] user hava pipe'permission,but can't use database
### Steps to reproduce the behavior (Required)
create user and grant
```
create database pipe_load_db;
create user user_pipe_load identified by '123456';
grant all on ALL PIPES in all DATABASES to user_pipe_load;
```
use new user connect SR
```
use pipe_load_db
```
image

image

### Expected behavior (Required)
use database successfully
### Real behavior (Required)
use database failed
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
mysql> select current_version();
+--------------------+
| current_version()  |
+--------------------+
| branch-3.2-02293a6 |
+--------------------+
1 row in set (0.00 sec)

### PR DESCRIPTION
**Pull request by murphyatwork**
Why I'm doing:

What I'm doing:
- User has any privilege on database should be able to `use database`, including pipe

Fixes #34294

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by satanson — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Authorizer.java [hunk 1]
```diff
@@ -278,39 +280,54 @@ public static void checkActionInDb(UserIdentity currentUser, Set<Long> roleIds,
...
+        AccessDeniedException lastExcepton = null;
+        for (AccessControlChecker checker : checkers) {
             try {
-                getInstance().getAccessControlOrDefault(catalogName)
-                        .checkAnyActionOnAnyTable(currentUser, roleIds, catalogName, db);
-            } catch (AccessDeniedException e2) {
-                if (CatalogMgr.isInternalCatalog(catalogName)) {
-                    try {
-                        getInstance().getAccessControlOrDefault(InternalCatalog.DEFAULT_INTERNAL_CATALOG_NAME)
-                                .checkAnyActionOnAnyView(currentUser, roleIds, db);
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Authorizer.java:299]: lose the logical ?

**Review by Seaven — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Authorizer.java [hunk 2]
```diff
@@ -278,39 +280,54 @@ public static void checkActionInDb(UserIdentity currentUser, Set<Long> roleIds,
...
-                                    .checkAnyActionOnAnyMaterializedView(currentUser, roleIds, db);
-                        } catch (AccessDeniedException e4) {
-                            getInstance().getAccessControlOrDefault(InternalCatalog.DEFAULT_INTERNAL_CATALOG_NAME)
-                                    .checkAnyActionOnAnyFunction(currentUser, roleIds, db);
-                        }
-                    }
-                } else {
-                    throw new AccessDeniedException();
-                }
+                checker.check();
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Authorizer.java:322]: for external catalog, we don't need to check mv, function, pipe

Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Authorizer.java:322]: it's checked in line 305

Comment (nshangyiming) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Authorizer.java:322]: table need to be checked for external catalog

Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Authorizer.java:317]: fixed

**Review by nshangyiming — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**